contrib/endpoints/src/api_manager/service_control/aggregated.cc

```diff
@@ -144,6 +177,13 @@ Aggregated::Aggregated(const ::google::api::Service& service,
         auth::ServiceAccountToken::JWT_TOKEN_FOR_SERVICE_CONTROL,
         url_.service_control() + servicecontrol_service);
   }
+
+  sa_token_quota_ = new auth::ServiceAccountToken(env);
+  sa_token_quota_->SetClientAuthSecret(
+      server_config->google_authentication_secret());
```

Comment:
You should NOT get token from server_file.  You could define a new type
 JWT_TOKEN_FOR_QUOTA_CONTROL, and set it audience.  then you can get its token.